robyn/scaffold/structured/no-db/devops/Dockerfile

```diff
@@ -0,0 +1,14 @@
+FROM python:3.11-bookworm AS builder
+
+WORKDIR /workspace
+
+COPY . .
+RUN pip install --no-cache-dir --upgrade -r requirements.txt --target=/workspace/deps
+
+FROM gcr.io/distroless/python3-debian11
```

Comment:
could you explain a bit more here?

Reply:
This is a two stage commit...

- takes python base image as builder
- copies source and does pip install in this stage 
- second stage is gcr distroless debian 11 (https://github.com/GoogleContainerTools/distroless) 
- distroless is a minimal image meant for production usage which is regularly maintained and patched for security vulnerabilities, and comes with an added benefit of small size as it doesn't include stuff from base linux image that's not needed at runtime
- we copy the site-packages dir, which is /workspace/deps in our case so that our dependencies are available
- finally, we set pythonpath and add server.py to the image's CMD